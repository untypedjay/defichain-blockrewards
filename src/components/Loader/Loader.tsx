import React from 'react';
import styled, { keyframes } from 'styled-components';

const loading = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

const StyledLoader = styled.div`
  display: flex;
  flex-direction: column;
  align-item: center;
  justify-content: center;
  
  @media (max-width: 830px) {
    margin-top: -20em;
    height: 100vh;
  }
`;

const StyledAnimation = styled.div`
  transition: all .25s ease;
  position: relative;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  
  &:after {
    content: '';
    position: absolute;
    border-radius: 50%;
    border: var(--anm-border) solid var(--clr-text);
    width: var(--anm-size);
    height: var(--anm-size);
    border-left: var(--anm-border) solid transparent;
    border-bottom: var(--anm-border) solid transparent;
    animation: ${loading} var(--anm-time) ease infinite;
    z-index: 2;
  }
  
  &:before {
    z-index: 1;
    content: '';
    position: absolute;
    border-radius: 50%;
    border: var(--anm-border) dashed var(--clr-text);
    width: var(--anm-size);
    height: var(--anm-size);
    border-left: var(--anm-border) solid transparent;
    border-bottom: var(--anm-border) solid transparent;
    animation: ${loading} var(--anm-time) linear infinite;
  }
`;

export default function Loader() {
  return (
    <StyledLoader>
      <StyledAnimation/>
    </StyledLoader>
  );
}