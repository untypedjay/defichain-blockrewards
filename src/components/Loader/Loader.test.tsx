import React from 'react';
import ReactDOM from 'react-dom';
import Loader from './Loader';
import { cleanup, render } from '@testing-library/react';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Loader/>, div);
});

it('renders correctly', () => {
  render(<Loader/>);
});

it('matches snapshot', () => {
  const { asFragment } = render(<Loader/>);
  expect(asFragment()).toMatchSnapshot();
});