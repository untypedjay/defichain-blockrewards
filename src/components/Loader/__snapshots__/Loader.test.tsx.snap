// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ gCaEYZ"
  >
    <div
      class="sc-gsTCUz kdtjXZ"
    />
  </div>
</DocumentFragment>
`;
